packages/shared/utils/common.ts

```diff
@@ -16,3 +28,71 @@ export const getGVK = (label: string) => {
 export const referenceForGroupVersionKind = (group: string) => (version: string) => (
     kind: string,
 ) => [group, version, kind].join('~');
+
+export const hasLabel = (obj: K8sResourceCommon, label: string): boolean =>
+  _.has(obj, ['metadata', 'labels', label]);
+
+export const getLabel = <A extends K8sResourceCommon = K8sResourceCommon>(
+    value: A,
+    label: string,
+    defaultValue?: string,
+  ) => (_.has(value, 'metadata.labels') ? value.metadata.labels[label] : defaultValue);
+
+export const getName = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.name') as K8sResourceCommon['metadata']['name'];
+
+export const getNamespace = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.namespace') as K8sResourceCommon['metadata']['namespace'];
+
+export const getUID = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.uid') as K8sResourceCommon['metadata']['uid'];
+
+export const getAnnotations = <A extends K8sResourceCommon = K8sResourceCommon>(
+    value: A,
+    defaultValue?: K8sResourceCommon['metadata']['annotations'],
+) => (_.has(value, 'metadata.annotations') ? value.metadata.annotations : defaultValue);
+
+export const resourcePathFromModel = (model: K8sKind, name?: string, namespace?: string) => {
+    const { plural, namespaced, crd } = model;
+  
+    let url = '/k8s/';
+  
+    if (!namespaced) {
+      url += 'cluster/';
+    }
+  
+    if (namespaced) {
+      url += namespace ? `ns/${namespace}/` : 'all-namespaces/';
+    }
+  
+    if (crd) {
+      url += referenceForModel(model);
+    } else if (plural) {
+      url += plural;
+    }
+  
+    if (name) {
+      // Some resources have a name that needs to be encoded. For instance,
+      // Users can have special characters in the name like `#`.
+      url += `/${encodeURIComponent(name)}`;
+    }
+  
+    return url;
+};
+
+export const isDefaultClass = (storageClass: K8sResourceKind) => {
+  const annotations = _.get(storageClass, 'metadata.annotations') || {};
+  return (
+    annotations[defaultClassAnnotation] === 'true' ||
+    annotations[betaDefaultStorageClassAnnotation] === 'true'
+  );
+};
+
+export const getLastLanguage = (): string => localStorage.getItem(LAST_LANGUAGE_LOCAL_STORAGE_KEY);
+
+export const k8sPatchByName = (
+  kind: K8sKind,
+  name: string,
+  namespace: string,
+  data: Patch[],
+) => k8sPatch({model: kind, resource: { metadata: { name, namespace } }, data: data}) as Promise<any>;
```

Comment:
```suggestion
) => k8sPatch({model: kind, resource: { metadata: { name, namespace } }, data: data}) as Promise<any>;
```
Add proper types for Promise.